src/Components/ModelledPropertyBuilder/ModelledPropertyBuilder.tsx

```diff
@@ -141,6 +145,13 @@ const ModelledPropertyBuilder: React.FC<ModelledPropertyBuilderProps> = ({
         [onChange]
     );
 
+    const onClickClearButton = useCallback(() => {
```

Comment:
nit: for on click handler naming convention the style is usually on....Click, or even you can use handleClearButtonClick: https://stackoverflow.com/questions/60048249/what-is-the-right-name-of-event-handler-onclick-or-handleclick
```suggestion
    const onClearButtonClick = useCallback(() => {
```

Reply:
+1 for the onClearButtonClick name